docs/getting_started/java/first_program_in_java/index.md

```diff
@@ -1,373 +1,757 @@
+<!--SNIPEND-->
+
+When you start a Worker, it begins polling the Task Queue for Tasks to process.
+The Worker continues running and blocks, waiting for Tasks to execute.
+It does not exit and return control to the command line.
+
+A Worker:
+
+- Only processes Workflow and Activity Tasks registered to it.
+- Knows which piece of code to execute based on the Tasks it gets from the Task Queue.
+- Only listens to the Task Queue that it's registered to.
 
-**Congratulations**, you just ran a Temporal Workflow application!
+### ![](/img/icons/harbor-crane.png) Use the Web UI to explore
 
-## ![](images/warning.png) Failure simulation
+Return to the [Temporal development service Web UI](http://127.0.0.1:8233).
+Now, there's one Worker registered.
+Before there was none.
+The Workflow status shows completion:
 
-So, you've just got a taste of one of Temporal's amazing value propositions: visibility into the Workflow and the status of the Workers executing the code. Let's explore another key value proposition, maintaining the state of a Workflow, even in the face of failures. To demonstrate this we will simulate some failures for our Workflow. Make sure your Worker is stopped before proceeding.
+![There is now one Worker and the Workflow is complete](images/completed-workflow.png)
+
+A complete Event History appears on the bottom of the page, with the most recent events first.
+Here you see the Workflow Execution start and complete.
+You also see the Activity and Workflow Tasks being scheduled, started, and completed.
+
+Here's what happens when the Worker ran:
+
+- The first Task the Worker finds is the one that tells it to execute the Workflow.
+- The Worker initiates the Workflow Execution by creating a new Workflow Task.
+- The Workflow Task initiates the Workflow method execution.
+- In turn, this method arrives at the Activity stubs. The first is 'Withdraw'.'
+- The Worker schedules the Activity Task to request Activity Execution.
+- The Activity Task starts and completes. The Worker communicates these events back to the Server.
+- Control returns to the Workflow by scheduling a new Workflow Task, allowing the Workflow Execution to continue to the next Activity, 'Deposit'.
+- The Worker schedules the new Activity Task, which is then picked up, started, and completed.
+- Again control goes back to the Workflow and a new Workflow Task.
+- The Worker schedules the Workflow task, picks it up for execution, starts it and completes it.
+- With this, the Workflow Execution completes as there is no more work to be done or Tasks to schedule.
+
+Each step is recorded in the Event History.
+
+Now that you've seen an Execution run and complete, it's time to introduce a failure.
+
+## ![](/img/icons/warning.png) Simulate failures
+
+Temporal is built to allow failure-prone processes to keep moving forward.
+It provides the backbone and oversight (called "Orchestration") that enables this without you having to develop the technology in your own code.
```

Comment:
We need to check if we're allowed to say Orchestration. We've gotten some flack about this before.

Reply:
👍